### DISCUSSION TITLE
metaplasmidSPAdes reconstructs identical plasmids

### DISCUSSION THREAD

**Discussion by barbaracania** [ablab/spades#968] [Q&A]
Hello!
I noticed that metaplasmidSPAdes reconstructed the exact same plasmid twice (I checked the sequences) using a different cutoff. The names of the identical contigs are: `NODE_1_length_65404_cov_23.803361_cutoff_10_type_circular` and `NODE_1_length_65404_cov_23.803361_cutoff_15_type_circular`. Is it an artifact or is there a reason for such behavior?
Best,
Barbara

**Comment by asl** [comment]
Tagging @Dmitry-Antipov – they author of metaplasmidSPAdes

**Comment by Dmitry-Antipov** [comment]
Hi.
Sorry for the delay. This looks like a bug. Can you share the data? it could simplify the fixing.

If not, can you check whether these sequences are identical or have some minor differences (i.e. circular shift or SNP) and attach spades.log file?

**Reply by barbaracania** [reply]
Hi,
Thank you for your answer! The original fastq files are too big to upload them here, so I attached only the spades.log file and the sequences of both plasmids.
Regarding the difference between the two plasmids, there is indeed a slight circular shift of a few hundred nucleotides between them, but otherwise one is a perfect reverse complement of the other, without a single SNP.

[spades.log](https://github.com/ablab/spades/files/8752348/spades.log)
[NODE_1_length_65531_cov_23.803361_cutoff_10_type_circular.txt](https://github.com/ablab/spades/files/8752352/NODE_1_length_65531_cov_23.803361_cutoff_10_type_circular.txt)
[NODE_1_length_65531_cov_23.803361_cutoff_15_type_circular.txt](https://github.com/ablab/spades/files/8752353/NODE_1_length_65531_cov_23.803361_cutoff_15_type_circular.txt)
r

**Reply by Dmitry-Antipov** [reply]
Thank you
Seems that we still need the reads to understand why this may happen. Can you share them with us (i.e through google drive, to spades.support@cab.spbu.ru)?

**Reply by Dmitry-Antipov** [reply]
Sorry for bothering you again, but still - could you send the data to us to reproduce the issue?

**Reply by barbaracania** [reply]
No problem at all, I am sorry myself that I did not contact you earlier with some info on when I would be able to share the data. I have uploaded the reads to google drive now and made them available for you. As for the parameters used, I ran spades only with the --metaplasmid flag, everything else was set to default.

**Reply by Dmitry-Antipov** [reply]
Got it, thank you!

**Reply by Dmitry-Antipov** [reply]
Thank you, now the reason of this problem is clear.
These duplication is an artifact (a bug, actually) ), no real biological reasons.

Hope to fix this till next release.

**Comment by asl** [comment]
The issue was addressed in https://github.com/ablab/spades/releases/tag/v3.15.5
